src/Integration.Vsix/SonarLintDaemonPackage.cs

```diff
@@ -78,16 +79,17 @@ public SonarLintDaemonPackage()
 
         #region Package Members
 
-        protected override Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            JoinableTaskFactory.RunAsync(InitAsync);
-            return Task.CompletedTask;
+            await JoinableTaskFactory.RunAsync(InitAsync);
         }
 
         private async Task InitAsync()
         {
             try
             {
+                await MigrateBindingsToServerConnectionsIfNeededAsync();
```

Comment:
This should be below the logger initialization and the initialization message.

```
[ThreadId 58] [Connection Migration] Start migrating connections from existing bindings
[ThreadId 58] [Connection Migration] The server connection with the id duncanp-sonar-test was not migrated as it already exists.
[ThreadId 58] Initializing the daemon package...
```
to 
```
[ThreadId 58] Initializing the daemon package...
[ThreadId 58] [Connection Migration] Start migrating connections from existing bindings
[ThreadId 58] [Connection Migration] The server connection with the id duncanp-sonar-test was not migrated as it already exists.
```